ruby/red-arrow/test/test-slicer.rb

```diff
@@ -484,4 +484,38 @@ def setup
 7	   256	true   
     TABLE
   end
+
+  sub_test_case "MatchSubstringOptions family" do
+    def setup
+      @string_table = Arrow::Table.new(
+        index: [*1..5],
```

Comment:
Why do we need this?

Reply:
Yes it's redundant. I feel something strange for the behavior of propagating nulls to other columns, so I put it here to show as example for users. But I will remove `:index` column here.

May be off topic of this PR, is it useful to propagate nulls in slicer?

I prefer to same behavior of `table#filter(boolean_array)` or `slicer.select { |value| .. }`.
(Null is not selected.)

Reply:
It's just for keeping backward compatibility.
See also: #6732

But we may can change it for 12.0.0. (2023-04-11 is code freeze date: https://lists.apache.org/thread/wj2x50s2xyghnjm0pm4gt7yq5187zhwy )
Could you open an issue and a pull request for it?

Reply:
FYI: The following change will work for it:

```diff
diff --git a/ruby/red-arrow/lib/arrow/table.rb b/ruby/red-arrow/lib/arrow/table.rb
index a14bf90d1f..da79e94b18 100644
--- a/ruby/red-arrow/lib/arrow/table.rb
+++ b/ruby/red-arrow/lib/arrow/table.rb
@@ -316,8 +316,6 @@ module Arrow
         end
       end
 
-      filter_options = Arrow::FilterOptions.new
-      filter_options.null_selection_behavior = :emit_null
       sliced_tables = []
       slicers.each do |slicer|
         slicer = slicer.evaluate if slicer.respond_to?(:evaluate)
@@ -339,7 +337,7 @@ module Arrow
           to += n_rows if to < 0
           sliced_tables << slice_by_range(from, to)
         when ::Array, BooleanArray, ChunkedArray
-          sliced_tables << filter(slicer, filter_options)
+          sliced_tables << filter(slicer)
         else
           message = "slicer must be Integer, Range, (from, to), " +
             "Arrow::ChunkedArray of Arrow::BooleanArray, " +
```

Reply:
I see. I will open an issue and PR for it. Thanks.